### DISCUSSION TITLE
Submission data for dynamic notification is incorrect after adding additional addresses between existing

### DISCUSSION THREAD

**Discussion by CAIDcreative** [solspace/craft-freeform#610] [Feature Requests]
**Describe the bug or issue you're experiencing**

We noticed that past submission data did not make sense for the dynamic notification field because the address of those submissions was only recently added while the submission had existed for much longer. Upon doing some testing it seems that the the data in the dynamic notification field submission only seems to save an integer of the position of the email rather than the value (which would be the email).

This actually causes us some pretty big issues because multiple new addresses have been added over time, and they have been placed between other address (because the ordering mattered) but now previous submissions seem muddled on where they were actually sent.

**Steps to reproduce**

1. Create a new dynamic recipient field with two options (test1@example.com, test2@example.com)
2. Add that field to a form
3. Submit that form with test2@example.com
4. Go back to the form and insert a third value (test3@example.com) between the other two so order is "test1, test3, test2"
5. The submission data we sent will say it was submitted to test3@example.com even though we really sent it to test2@example.com

**Expected behavior**

Should always store the email regardless of how the field changes

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 3.7.63.1
- Freeform Version: Pro
- Freeform Edition: 3.13.22.1
- Fresh Install or Upgrade: Upgrade

**Comment by kjmartens** [comment]
I'm really sorry about the issue you're experiencing @CAIDcreative. Unfortunately, this is an architectural issue that we cannot resolve on v3 and v4. However, we have already addressed this issue for Freeform 5 (available in several months).

**Comment by kjmartens** [comment]
Email notifications have been overhauled in Freeform 5 for Craft 4 (now in beta), and Dynamic Recipients are now handled differently as a "layer" upon regular field types, resolving this issue. 🙂
